### DISCUSSION TITLE
Env variables not usable during command line (knex migrate)

### DISCUSSION THREAD

**Discussion by kseanjbz** [vercel/next.js#14850] [Help] [answered]
My knexfile refers to an env var in .env.local

This is not a problem with the website running on dev. But when I try to execute a command line command such as knex migrate:latest which uses the same knexfile the environment variable is no longer being picked up. 

What is the best way to handle this?

**Comment by rafaelalmeidatk** [comment, marked as answer]
You are executing the knex CLI directly, it doesn't load the `.env` file automatically. You would have to use `dotenv` in your knexfile to load your variables:
```js
require('dotenv').config({ path: '.env.local' });

module.exports = {
  client: 'pg',
  connection: {
    ...
  }
};
```

**Comment by kseanjbz** [comment]
Yeah I was hoping there would be a built in solution so I don't need to use dotenv. Thanks for the help!

**Comment by advename** [comment]
A better solution would be to use Next's built-in package, [`@next/env`](https://nextjs.org/docs/basic-features/environment-variables#test-environment-variables)

E.g.
```js
const { loadEnvConfig } = require("@next/env");
loadEnvConfig(".") // "." is for the directory where the .env, .env.local, ... is located. Follows Next.js environment loading order

// Environment variables are now available in this file, even in the cli, without any additional library.

module.exports = {
  client: process.env.DB_CLIENT,
  connection: {
    ...
  }
};
```
